kura/test/org.eclipse.kura.container.orchestration.provider.test/src/test/java/org/eclipse/kura/container/orchestration/provider/EnforcementSecurityTest.java

```diff
@@ -0,0 +1,254 @@
+                .setEnvVars(Arrays.asList("test=test", "test2=test2")).build();
+
+        doReturn(containerDescriptors).when(this.mockedContainerOrchImpl).listContainerDescriptors();
+
+        doNothing().when(this.mockedContainerOrchImpl).pullImage(any(ImageConfiguration.class));
+    }
+
+    private void givenMockedDockerClient(String[] digestsList, String containerId, String containerName,
+            String imageName) {
+        DockerClient mockedDockerClient = mock(DockerClient.class, Mockito.RETURNS_DEEP_STUBS);
+        List<Image> images = new LinkedList<>();
+        Image mockImage = mock(Image.class);
+
+        when(mockImage.getRepoTags()).thenReturn(new String[] { imageName, "latest", "nginx:latest" });
+        when(mockImage.getRepoDigests()).thenReturn(digestsList);
+        when(mockImage.getId()).thenReturn(imageName);
+        images.add(mockImage);
+
+        when(mockedDockerClient.listImagesCmd()).thenReturn(mock(ListImagesCmd.class));
+        when(mockedDockerClient.listImagesCmd().withImageNameFilter(anyString())).thenReturn(mock(ListImagesCmd.class));
+        when(mockedDockerClient.listImagesCmd().withImageNameFilter(anyString()).exec()).thenReturn(images);
+        when(mockedDockerClient.stopContainerCmd(anyString())).thenReturn(mock(StopContainerCmd.class));
+        when(mockedDockerClient.stopContainerCmd(anyString()).exec()).thenAnswer(answer -> {
+            this.stoppingResult = ContainerState.STOPPING;
+            return null;
+        });
+
+        when(mockedDockerClient.removeContainerCmd(anyString())).thenReturn(mock(RemoveContainerCmd.class));
+        when(mockedDockerClient.removeContainerCmd(anyString()).exec()).thenAnswer(answer -> {
+            this.containerInstanceDescriptor = ContainerInstanceDescriptor.builder().setContainerID(containerId)
+                    .setContainerName(containerName).setContainerImage(imageName)
+                    .setContainerState(ContainerState.FAILED).build();
+            return null;
+        });
+
+        this.mockedContainerOrchImpl.setDockerClient(mockedDockerClient);
+    }
+
+    private void givenAllowlistEnforcement(String rawAllowlistContent) {
+        this.allowlistEnforcementMonitor = new AllowlistEnforcementMonitor(rawAllowlistContent,
+                this.mockedContainerOrchImpl);
+    }
+
+    /*
+     * When
+     */
+
+    private void whenOnNext(String containerId) {
+        this.allowlistEnforcementMonitor.onNext(new Event("start", containerId, "nginx:latest", 1708963202L));
+    }
+
+    private void whenVerifyAlreadyRunningContainersDigests(List<ContainerInstanceDescriptor> containerDescriptors) {
+        this.allowlistEnforcementMonitor.enforceAllowlistFor(containerDescriptors);
+    }
+
+    /*
+     * Then
+     */
+
+    private void thenContainerDigestIsVerified() {
+
+        assertEquals(ContainerState.ACTIVE, this.containerInstanceDescriptor.getContainerState());
+
+    }
```

Comment:
... on another note you could simplify these checks a lot by simply passing the expected state

```suggestion
    private void thenContainerIs(ContainerState expectedContainerState) {
        assertEquals(expectedContainerState, this.containerInstanceDescriptor.getContainerState());
    }
```